src/NuGetPackageVerifier/Rules/PackageVersionMatchesAssemblyVersionRule.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Mono.Cecil;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class PackageVersionMatchesAssemblyVersionRule : AssemblyHasAttributeRuleBase
+    {
+        private Version _packageVersion;
+        private string _packageId;
+
+        public override IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
+        {
+            _packageVersion = context.Metadata.Version.Version;
+            _packageId = context.Metadata.Id;
+            return base.Validate(context);
+        }
+
+        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(string currentFilePath,
+            Mono.Collections.Generic.Collection<CustomAttribute> assemblyAttributes)
+        {
+            var versionAttribute = assemblyAttributes.SingleOrDefault(a =>
+                a.AttributeType.FullName.Equals(
+                    typeof(AssemblyInformationalVersionAttribute).FullName ??
```

Comment:
Should it instead only look at `AssemblyInformationalVersionAttribute`? The reason I did this was because according to NuGet's [docs](https://docs.microsoft.com/en-us/nuget/schema/nuspec#Replacement_Tokens) if `AssemblyInformationalVersionAttribute` is not specified, the `$version$` value defaults to `AssemblyVersionAttribute`.